mephisto/abstractions/_subcomponents/agent_state.py

```diff
@@ -127,7 +127,7 @@ def set_init_state(self, data: Any) -> bool:
         raise NotImplementedError()
 
     @abstractmethod
-    def get_init_state(self) -> Optional[Any]:
+    def get_init_state(self, get_all_state: Optional[bool] = False) -> Optional[Any]:
```

Comment:
This is for the static_agent_state.py 's implementation of get_init_state(). For some reason the default version of this method only returned state["inputs"], which is not enough as state["metadata"] is also needed. Setting this property to true allows for the whole state object to be returned.

Reply:
got it, thanks for the explanation, I understand the need here. I'd like to get @JackUrb 's thoughts on this as well.

The overloaded fn signature feels a little awkward to me, I feel as if we should have a separate abstract method instead so both `get_init_state` and `get_full_state` or something. We would then have to implement that abstract method though in all the derived classes. Would also have to understand if this approach would make the upgrade story complicated for users.

Reply:
Oof this is a bit hairy and points to some of my pending refactors here. `StaticAgentState` for instance should be returning the whole state object as well, and, in general, `AgentState` should have some top-level attributes that are shared and known.

I'd consider this low-hanging fruit for a code quality refactor (in 2 stages). The base `AgentState` class should define the attributes we expect to be in _every_ task, and then in the subclasses that have inherited and perhaps used different names we can alias to the correct values with the `@property` decorator. Then we can mark those as deprecated and remove them in 1.2 or later.

Reply:
I don't want to block you on this for too long @Etesam913 - won't get a chance today but I can throw together the `AgentState` refactor tomorrow morning, then this would just need a rebase.

Reply:
It should be fully merged with your refactor now

Reply:
No it is not, just removed it